### ISSUE TITLE
Concerning parsing prediction 

### ISSUE THREAD

**Issue by Daniellli** [nickgkan/beauty_detr#40]
Hi, thank you for your wonderful work. 

I am trying to parse the prediction, and i feel confusing   at the following lines.
https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/train_dist_mod.py#L227-L232

 Why  replacing the last_sem_cls_scores to the new generated tokenidx and wordidx based class info? what do the tokenidx and wordidx mean? 

thank you for your attention

**Comment by Daniellli** [comment]
moreover, does this line right? 


dividing the probability of being object, the top class token with  highest response will be changed, maybe become the second largest one? 
https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/models/ap_helper.py#L150C20-L150C20

**Comment by ayushjain1144** [comment]
`Why replacing the last_sem_cls_scores to the new generated tokenidx and wordidx based class info?` : The idea is to convert the logits over the detection prompt span (i.e. 256 dimensions) to logits over class labels (18 classes and thus 19 dimensions to include no object class). 

`what do the tokenidx and wordidx mean?`: See #17 

`moreover, does this line right?`: This line ends up being inconsequential because [here](https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/models/ap_helper.py#L222) we multiply back sem_cls_scores with obj_prob. We will fix this in the next version.

**Comment by Daniellli** [comment]
Hi,  

does the [line 153 ](https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/utils/nms.py#L153) correct? if it is correct, the boxes with high IoU but diffferent prediction class will be maintained, which i think a little unreasonable if the IoU is extremely high, such as 1?

**Comment by Daniellli** [comment]
moreover, the [line 155](https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/utils/nms.py#L155)  make me confusing, why the [last - 1] are always excluded?  is there any probability about the [last -1] box is the  correctest one?

**Comment by ayushjain1144** [comment]
`nms.py` comes unchanged from [votenet](https://github.com/facebookresearch/votenet/blob/main/utils/nms.py), I believe they would be able to better assist you. We didn't really look deep into the nms code.

**Comment by Daniellli** [comment]
hi , i am little confusing with following lines:
https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/src/grounding_evaluator.py#L197-L201


why  the GT involved in parsing prediciton ?
